crates/turborepo-lib/src/shim.rs

```diff
@@ -176,50 +178,234 @@ struct PackageJson {
     version: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct LocalTurboState {
-    bin_path: PathBuf,
-    version: String,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct YarnRc {
+    pnp_unplugged_folder: PathBuf,
 }
 
-impl LocalTurboState {
-    pub fn infer(repo_root: &Path) -> Option<Self> {
-        let local_turbo_path = repo_root.join("node_modules").join(".bin").join({
+impl Default for YarnRc {
+    fn default() -> Self {
+        Self {
+            pnp_unplugged_folder: [".yarn", "unplugged"].iter().collect(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TurboState {
+    bin_path: Option<PathBuf>,
+    version: &'static str,
+    repo_state: Option<RepoState>,
+}
+
+impl Default for TurboState {
+    fn default() -> Self {
+        Self {
+            bin_path: env::current_exe().ok(),
```

Comment:
Do we want to error if we can't get `current_exe`?

Reply:
That can fail for lots of reasons, most particularly permissions. It's not an error to not know who we are, so I made it an `Option` instead.

The only thing it means is that we can't prevent invoking inner ("local") `turbo` if it just so happens to be the same one. But! The thing that would _prevent_ us from discovering `current_exe` would almost guarantee that we're not going to discover it when we go through looking for a `turbo` to delegate to.